File: app/src/models/UserInteraction.js
const mongoose = require('mongoose');
const mongoosePaginate = require('mongoose-paginate');

const UserInteractionSchema = new mongoose.Schema({
    website: {
        type: String,
        required: true
    },
    event: {
        type: String,
        required: true
    },
    user: {
        type: String,
        required: true
    },
    timestamp: {
        type: Date,
        default: Date.now
    }
});

UserInteractionSchema.plugin(mongoosePaginate);
mongoose.model('UserInteraction', UserInteractionSchema);

File: gateway-api/src/main/java/com/inloco/gateway/model/UserInteraction.java
package com.inloco.gateway.model;

import java.util.Date;

public class UserInteraction {

	private String website;
	private String user;
	private InteractionType event;
	private Date timestamp;

	public String getWebsite() {
		return website;
	}

	public void setWebsite(String website) {
		this.website = website;
	}

	public String getUser() {
		return user;
	}

	public void setUser(String user) {
		this.user = user;
	}

	public InteractionType getEvent() {
		return event;
	}

	public void setEvent(InteractionType event) {
		this.event = event;
	}

	public Date getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(Date timestamp) {
		this.timestamp = timestamp;
	}

}
